### PR TITLE
2713 block view alter

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/2713

### CODE REVIEW

File: core/modules/layout/plugins/renderers/layout_renderer_standard.inc [hunk 1]
```diff
@@ -120,21 +120,6 @@ class LayoutRendererStandard {
...
-   * setting or getting the title within a single request.
-   *
-   * This value is not escaped and should be run through check_plain() when
-   * adding to the page. The title should already be translated when set.
-   *
-   * @var string
-   */
-  protected $page_title = NULL;
-
-  /**
```
Comment (quicksketch) [inline core/modules/layout/plugins/renderers/layout_renderer_standard.inc]: These changes seem unrelated.

File: core/modules/layout/plugins/renderers/layout_renderer_standard.inc [hunk 2]
```diff
@@ -523,12 +483,19 @@ class LayoutRendererStandard {
...
         'content' => $content,
       );
 
+      // Add a backwards compatible way to alter all blocks.
+      backdrop_alter(array('block_view', 'block_view_' . $block->module . '_' . str_replace('-', '_', $block->delta)), $data, $block);
+
+      $content = (object) array(
+        'title' => $data['subject'],
+        'content' => $data['content'],
+      );
```
Comment (quicksketch) [inline core/modules/layout/plugins/renderers/layout_renderer_standard.inc]: The comment `// Add a backwards compatible way to alter all blocks.` should remove "backwards compatible", as this is actually an explicit API change from Drupal 7.

As this is an API change, perhaps we should change the `$data` that is passed into `hook_block_alter()` to use "title" instead of "subject"? Then *only* legacy blocks using `hook_block_view()` would still be using "subject".

**Review by quicksketch — DISMISSED** [review]
Overall the PR in concept is great! There's a completely unrelated change to page titles in here though, could you update?

**Comment by jenlampton** [comment]
Removed the accidental changes to layout tiles and pushed a new commit.

File: core/modules/layout/plugins/renderers/layout_renderer_standard.inc [hunk 3]
```diff
@@ -523,10 +523,18 @@ class LayoutRendererStandard {
...
 
+      $data = array(
+        'subject' => $title,
+        'content' => $content,
+      );
+
+      // Add a backwards compatible way to alter all blocks.
+      backdrop_alter(array('block_view', 'block_view_' . $block->module . '_' . str_replace('-', '_', $block->delta)), $data, $block);
+
       // Assemble a structured content array.
```
Comment (opi) [inline core/modules/layout/plugins/renderers/layout_renderer_standard.inc]: s/array/object/ ?

**Review by opi — APPROVED** [review]
untested, but LGTM

**Comment by quicksketch** [comment]
Merged into 1.x for 1.8.0.
